### ISSUE TITLE
timezone handling broken, stores epoch with timezone offset in postgres database

### ISSUE THREAD

**Issue by tik-stbuehler** [Icinga/icinga2#6584]
When querying `next_check` from the postgresq database I get the timestamp in the local timezone (UTC+2 right now).  The API returns the correct timestamp (without timezone offset).

Icingaweb2 expects the timestamp in UTC in the database (as it should be, as it doesn't include the timezone anymore since #4791); it adds my local timezone offset again, so I get really broken times ("UTC+4") in the web frontend.

## Expected Behavior

Icingaweb2 should show the actual timestamp in local time.

In postgres I should see timestamps in UTC.

## Current Behavior

Postgres data for a single service:

```
icinga2=# select next_check FROM icinga_servicestatus WHERE service_object_id=360;                                                                                                           
     next_check
---------------------
 2018-09-03 10:03:26
(1 row)

icinga2=# select UNIX_TIMESTAMP(next_check) FROM icinga_servicestatus WHERE service_object_id=360;                                                                                           
 unix_timestamp
----------------
     1535969006
(1 row)
```

In the API output I see (this is the actual UTC timestamp and correct):

```
"next_check" : 1535961806.88442,
```

Icingaweb2 shows:

```
<span class="relative-time time-until" title="2018-09-03 12:03:26">at 12:03</span>
```

## Your Environment

* Version: `2.6.0-2` from debian stretch
* Version used (`icinga2 --version`): `r2.6.0-1`

**Comment by Crunsher** [comment]
We can probably force postgres to read and store our timestamps as UTC.

**Comment by tik-stbuehler** [comment]
* https://github.com/Icinga/icinga2/blob/f0c9098a287e800e949d7a9f97daf6507ea536e7/lib/db_ido_pgsql/idopgsqlconnection.cpp#L676 looks fine
* https://github.com/Icinga/icinga2/blob/f0c9098a287e800e949d7a9f97daf6507ea536e7/lib/db_ido_pgsql/idopgsqlconnection.cpp#L679 is broken; `NOW()` uses local timezone by default
* https://github.com/Icinga/icinga2/blob/f0c9098a287e800e949d7a9f97daf6507ea536e7/lib/db_ido_pgsql/idopgsqlconnection.cpp#L950 is broken too (`TO_TIMESTAMP` default to local timezone)

**Comment by dnsmichi** [comment]
2.6.0 does not contain the `AT TIME ZONE 'UTC'` entry. You're using an old outdated version of Icinga 2 here which doesn't write UTC timestamps.

**Comment by tik-stbuehler** [comment]
So you managed to fix one out of three in #4874 - what feedback do you need now?

I'm not interested in updating to a more recent release, as the issue is not quite severe enough (especially given there is a workaround) to justify moving to debian testing or latest upstream.

**Comment by dnsmichi** [comment]
I'm saying that a more recent version would fix your problem with inserting the wrong timestamps into the database. So far, this is your best bet.

**Comment by tik-stbuehler** [comment]
No; the current version is still broken (lines 679 and 950 I quoted above), and my best bet is using UTC as timezone for postgres.

I also get that line 676 is probably the most used line, and updating would "fix" the problem I'm seeing.  Even when I'm ignoring the other two places updating is still not my best bet, because new version also means new bugs, and I like getting security updates for the software I'm using without having to use git HEAD.

My main question is still: what feedback do you need from me now? Because I'm guessing the `need-feedback` label is somehow meant for me.

**Comment by dnsmichi** [comment]
The only bug I can see here is that the cleanup queries miss the offset. `ToTimeStamp` is never used anywhere in the code, so the problem asked in this issue is mainly about timestamp inserts. You can wait until the cleanup queries are fixed, or use a different workaround.

**Comment by tik-stbuehler** [comment]
You again didn't answer my question, so I'm gonna stop responding here.

**Comment by dnsmichi** [comment]
I'm not interested in your tone either, so I'll just fix the problems and close here then.
